pkg/apis/identifiers.go

```diff
@@ -51,15 +50,18 @@ func SanitizeLabel(name string) string {
 // SanitizeName ensures a value is suitable for usage as an apiserver identifier.
 func SanitizeName(name string) string {
 	sanitized := name
+
 	if len(path.IsValidPathSegmentName(name)) != 0 {
 		for _, invalidName := range path.NameMayNotBe {
 			if name == invalidName {
 				// the only strictly invalid names are `.` and `..` so this is sufficient
 				return strings.ReplaceAll(name, ".", "_")
 			}
 		}
-		sanitized = invalidPathCharacters.ReplaceAllString(sanitized, "_")
```

Comment:
I realize after the fact that this alone is a pretty significant change. I don't believe it's inherently a *problem*, as a URL escape will still produce valid path names, but I think a less broad change would be to replace `%` with `_` as the original implementation did, and replace `/` with `%2F`.

Reply:
This function is called against extension module names and you lose context: `module/sub` turns into `module_sub` which could be the name of an actual extension.

Since `%2F` is valid in path names it seemed a reasonable compromise.